internal/redfishwrapper/system.go

```diff
@@ -35,7 +35,12 @@ func (c *Client) Systems() ([]*redfish.ComputerSystem, error) {
 		return nil, err
 	}
 
-	return c.client.Service.Systems()
+	s, err := c.client.Service.Systems()
+	if err != nil {
+		return nil, err
+	}
+
+	return c.matchingSystem(s), nil
```

Comment:
We'd want to keep the default behavior here - to return all Systems - when c.systemName was not specified

```suggestion
        if c.systemName != "" {
	    return c.matchingSystem(s), nil
	}
	
	return s, nil
```